util/pace/util/monitor/netcdf_monitor.py

```diff
@@ -0,0 +1,174 @@
+import logging
+from pathlib import Path
+from typing import Any, Dict, List, Optional, Set
+
+import fsspec
+import numpy as np
+
+from pace.util.communicator import CubedSphereCommunicator
+
+from .. import _xarray as xr
+from ..filesystem import get_fs
+from ..quantity import Quantity
+
+
+logger = logging.getLogger(__name__)
+
+
+class _TimeChunkedVariable:
+    def __init__(self, initial: Quantity, time_chunk_size: int):
+        self._data = np.zeros(
+            (time_chunk_size, *initial.extent), dtype=initial.data.dtype
+        )
+        self._data[0, ...] = initial.view[:]
+        self._dims = initial.dims
+        self._units = initial.units
+        self._i_time = 1
+
+    def append(self, quantity: Quantity):
+        self._data[self._i_time, ...] = quantity.transpose(self._dims).view[:]
```

Comment:
I left it out on this one because I don't recall a time in three years that this check has ever caused an exception to be raised. Even if there are extra or fewer dims, it should be possible enough to tell a dimension got added or removed.